SofaKernel/modules/Sofa.Type/src/sofa/type/Vec.h

```diff
@@ -422,12 +422,20 @@ class Vec : public sofa::type::fixed_array<ValueType,size_t(N)>
     }
 
     /// Vector negation.
-    constexpr Vec<N,ValueType> operator-() const noexcept
+    constexpr Vec<N, ValueType> operator-() const noexcept
     {
-        Vec<N,ValueType> r(NOINIT);
-        for (Size i=0; i<N; i++)
-            r[i]=-this->elems[i];
-        return r;
+        if constexpr (std::is_unsigned_v<ValueType>)
+        {
+            static_assert("Trying to negate an unsigned vector");
```

Comment:
Should we stop the compilation if we try to negate a vector of unsigned ?

Reply:
Yes, that's the idea